cabal-install/src/Distribution/Client/Init/NonInteractive/Heuristics.hs

```diff
@@ -159,10 +159,15 @@ guessAuthorEmail = guessGitInfo "user.email"
 
 guessGitInfo :: Interactive m => String -> m String
 guessGitInfo target = do
-  info <- readProcessWithExitCode "git" ["config", "--local", target] ""
-  if null $ snd' info
-    then trim . snd' <$> readProcessWithExitCode "git" ["config", "--global", target] ""
-    else return . trim $ snd' info
+  localInfo <- readProcessWithExitCode "git" ["config", "--local", target] ""
+  if null $ snd' localInfo
+    then do
+      globalInfo <- readProcessWithExitCode "git" ["config", "--global", target] ""
+      case fst' globalInfo of
+        ExitSuccess -> return . trim $ snd' globalInfo
+        _           -> return ""
```

Comment:
Perhaps, more prudent would be to return `Nothing` (changing the type to `m (Maybe String)`)?

Reply:
Yeah that's probably a good idea. I have to see how that affects the prompt functions that use this. Would you agree that passing `MandatoryPrompt` instead of `DefaultPrompt` is a good idea in the case of `Nothing` like I outlined above?

Reply:
```
emailPrompt :: Interactive m => InitFlags -> m String
emailPrompt flags = getEmail flags $ do
    email' <- guessAuthorEmail
    case email' of
      Nothing -> promptStr "Maintainer email" MandatoryPrompt
      Just e  -> promptStr "Maintainer email" (DefaultPrompt e)
```

Such as this I mean

Reply:
Yeah, I don't remember the whole API by heart and would need to double check (which I don't have time for right now, unfortunately), but on the surface of it, it looks legit.

Reply:
I added Maybe and resolved the new type for `src/Distribution/Client/Init/Interactive/Command.hs`, but in `src/Distribution/Client/Init/NonInteractive/Command.hs` in the functions `authorHeuristics` and `emailHeuristics` these guessing functions are used as well, so the type needs to be adjusted there as well.

However, I'm having trouble seeing how to do this / how these functions work.